org.adempiere.ui.zk/WEB-INF/src/org/adempiere/webui/dashboard/DPDocumentStatus.java

```diff
@@ -57,8 +59,15 @@ public class DPDocumentStatus extends DashboardPanel implements EventListener<Ev
 
 	@Override
 	public void refresh(ServerPushTemplate template) {
-		statusPanel.refresh();
-		template.executeAsync(this);
+		try {
+			Executions.activate(template.getDesktop());
+			statusPanel.refresh();
+			template.executeAsync(this);
+		} catch (DesktopUnavailableException | InterruptedException e) {
+			e.printStackTrace();;
+		} finally {
+			Executions.deactivate(template.getDesktop());

```

Comment:
this is highly not recommended.

Reply:
Thanks @hengsin 

The issue is because I needed to change the visibility of the indicator.

When the code indicator.setVisible was reached, an error was throws stating that the component can only be accessed on listeners.  I think that's caused because we're running the refresh on a non-UI thread.

Googling around the solution I found in stackoverflow (or zk forums) was to use the Executions.activate/deactivate

If that's not a good idea, how could we solve the issue, to change visibility and color running on non-UI thread?

Reply:
you should do that in the updateUI method instead. Using activate/deactivate means the whole UI is lock while the refresh method is executing.